src/agent_utils.rs

```diff
@@ -0,0 +1,192 @@
+use crate::err::Error;
+
+use crate::hooks::IS_VISIBLE_FUNC;
+use crate::hooks::TRACE_LINE_FUNC;
+
+pub static mut PLAYER1_REF: Option<&Playerent> = None;
+pub static mut PLAYER1: Option<u64> = None;
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone, Default)]
+pub struct Vec3 {
+    pub x: f32,
+    pub y: f32,
+    pub z: f32,
+}
+
+#[repr(C)]
+pub union world_pos {
```

Comment:
Follow Rust naming conventions
```suggestion
pub union WorldPos {
```